python/cudf/cudf/core/indexing_utils.py

```diff
@@ -145,7 +145,8 @@ def destructure_dataframe_iloc_indexer(
     rows, cols = destructure_iloc_key(key, frame)
     if cols is Ellipsis:
         cols = slice(None)
-    scalar = is_integer(cols)
+    elif isinstance(cols, (cudf.Series, cudf.Index)):
```

Comment:
I remember us removing this feature after explicitly raising errors for few releases.

Reply:
I'm afraid that I don't recall the history here. Could you point us, Prem? Presumably we disabled this for performance reasons?

Reply:
I'm not able to find the PR where we removed it. We did remove support for cudf columns to be passed as parameters for some of the APIs, but this fix seems appropriate so approving it. We can revisit if this causes perf issues later.